pages/habit-lists/[id].js

```diff
@@ -13,14 +11,16 @@ cardColors[2] = '#faf9e5';
 cardColors[3] = '#f7edf5';
 
 
-const List = () => {
+export default function List() {
   const router = useRouter()
   const { asPath } = useRouter();
   const { id } = router.query;
 
   const [habits, setHabits] = useState([]);
   const [url, setUrl] = useState('');
   const [effect, setEffect] = useState(false);
+  const [shareButtonText, setShareButtonText] = useState('Share List');
+  const [message, setMessage] = useState('');
 
   const fetchHabits = async () => {
     const habits = await fetch('/api/getHabits');
```

Comment:
I would expect us to filtering habits by the listID in the backend, rather than returning *all* habits and filtering in the frontend (like you're doing [here](https://github.com/farhaannishtar/Habit-Tracker/pull/8/files#diff-471239b0eb460c543ddd63898d3150efb81efc413493d1d2e9baad5c4097f58aR54-R55))

If a million users were using your site, you wouldn't want to be fetching *all* the habits everytime. We should update this so that we are only loading the habits for a particular listID in the backend/at the database layer.

Reply:
got it understood

Reply:
Are you planning to make that change in a separate PR?